### PR TITLE
Switchport Module Additions

### LINKED ISSUE #227: meraki_switchport access_policy_types choices don't match.
The Ansible Module has choices:
    access_policy_type:
        choices: [Open, Custom access policy, MAC whitelist, Sticky MAC whitelist].
        
 When a playbook uses this keyword, it will fail.
 
     "msg": "HTTP error 400 - https://api.meraki.com/api/v1/devices/Q2HP-HMV3-8PZW/switch/ports/6 - 'accessPolicyType' must be one of: 'Open', 'Custom access policy', 'MAC allow list' or 'Sticky MAC allow list'",
    "response": "OK (unknown bytes)",
    "status": 400

Looking at API docs, the valid options are listed in the schema.
https://developer.cisco.com/meraki/api-v1/#!update-device-switch-port

accessPolicyType:
string
The type of the access policy of the switch port. Only applicable to access ports. Can be one of 'Open', 'Custom access policy', 'MAC allow list' or 'Sticky MAC allow list'.

Fix is to change the `choices` in the arg_spec to the accurate options, which are 'Open', 'Custom access policy', 'MAC allow list' or 'Sticky MAC allow list'.

### PR DESCRIPTION
**Pull request by jeffkala**
fixes #227 

Adding mac allow list functionality, along with sticky port configuration options.  Also fixes choices for access_policy_types to align with schema on API documentation.

### CODE REVIEW

**Comment by kbreit** [comment]
@jeffkala I will work on a patch to fix the errors. In the meantime, are you able to write some integration tests for this? If you have an `ansible-test` integration test environment setup, you can use that. If not, contact me on Slack and I can help walk you through the setup.

**Comment by jeffkala** [comment]
> @jeffkala I will work on a patch to fix the errors. In the meantime, are you able to write some integration tests for this? If you have an `ansible-test` integration test environment setup, you can use that. If not, contact me on Slack and I can help walk you through the setup.

Submitted some tests

File: plugins/modules/meraki_ms_switchport.py [hunk 1]
```diff
@@ -109,7 +109,55 @@
         - VLAN number assigned to a port for voice traffic.
         - Only applicable to access port type.
         type: int
-
+    mac_allow_list:
+        description:
+        - MAC addresses list that are white listed(allowed) on a port.
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:114]: Please remove white list and only say allowed list instead.

File: plugins/modules/meraki_ms_switchport.py [hunk 2]
```diff
@@ -109,7 +109,55 @@
...
+                choices: [merged, replaced, deleted]
+                default: replaced
+            macs:
+                description:
+                - List of MAC addresses to update with based on state option.
+                type: list
+                elements: str
+    sticky_mac_allow_list:
+        description:
+        - MAC addresses list that are white listed(allowed) on a port.
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:135]: Please remove white list and only say allowed list instead.

File: plugins/modules/meraki_ms_switchport.py [hunk 3]
```diff
@@ -109,7 +109,55 @@
...
+            macs:
+                description:
+                - List of MAC addresses to update with based on state option.
+                type: list
+                elements: str
+    sticky_mac_allow_list_limit:
+        description:
+        - The number of MAC addresses allowed in the sticky port allow list.
+        - Only applicable to access port type.
+        - Only applicable to access_policy_type "Sticky MAC whitelist".
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:158]: Please remove white list and only say allowed list instead.

File: plugins/modules/meraki_ms_switchport.py [hunk 4]
```diff
@@ -109,7 +109,55 @@
...
+            macs:
+                description:
+                - List of MAC addresses to update with based on state option.
+                type: list
+                elements: str
+    sticky_mac_allow_list:
+        description:
+        - MAC addresses list that are white listed(allowed) on a port.
+        - Only applicable to access port type.
+        - Only applicable to access_policy_type "Sticky MAC whitelist".
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:137]: Please remove white list and only say allowed list instead.

File: plugins/modules/meraki_ms_switchport.py [hunk 5]
```diff
@@ -109,7 +109,55 @@
         - VLAN number assigned to a port for voice traffic.
         - Only applicable to access port type.
         type: int
-
+    mac_allow_list:
+        description:
+        - MAC addresses list that are white listed(allowed) on a port.
+        - Only applicable to access port type.
+        - Only applicable to access_policy_type "MAC whitelist".
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:116]: Please remove white list and only say allowed list instead.

File: plugins/modules/meraki_ms_switchport.py [hunk 6]
```diff
@@ -183,6 +231,45 @@
       - 15
       - 20
   delegate_to: localhost
+
+- name: Configure access port with sticky mac allow list and limit.
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:235]: Capitalize MAC

File: plugins/modules/meraki_ms_switchport.py [hunk 7]
```diff
@@ -183,6 +231,45 @@
...
+    state: present
+    serial: ABC-123
+    number: 5
+    sticky_mac_allow_list:
+        macs:
+          - aa:aa:bb:bb:cc:cc
+        state: deleted
+    delegate_to: localhost
+
+- name: Add a MAC address to sticky mac allow list.
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:262]: Capitalize MAC

File: plugins/modules/meraki_ms_switchport.py [hunk 8]
```diff
@@ -183,6 +231,45 @@
...
+    sticky_mac_allow_limit: 3
+    sticky_mac_allow_list:
+        macs:
+          - aa:aa:bb:bb:cc:cc
+          - bb:bb:aa:aa:cc:cc
+          - 11:aa:bb:bb:cc:cc
+        state: replaced
+    delegate_to: localhost
+
+- name: Delete an existing MAC address from the sticky mac allow list.
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:250]: Capitalize MAC

File: plugins/modules/meraki_ms_switchport.py [hunk 9]
```diff
@@ -256,6 +343,21 @@
             returned: success
             type: str
             sample: "Auto negotiate"
+        sticky_mac_allow_list_limit:
+            description: Number of MAC addresses allowed on a sticky port.
+            returned: success
+            type: int
+            sample: 6
+        sticky_mac_allow_list:
+            description: List of MAC addresses currently allowed on a sticky port. Used with access_policy_type of Sticky MAC whitelist.
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:352]: Please remove white list and only say allowed list instead.

File: plugins/modules/meraki_ms_switchport.py [hunk 10]
```diff
@@ -256,6 +343,21 @@
...
+            returned: success
+            type: list
+            sample: ["11:aa:bb:bb:cc:cc", "22:aa:bb:bb:cc:cc", "33:aa:bb:bb:cc:cc"]
+        mac_allow_list:
+            description: List of MAC addresses currently allowed on a non-sticky port. Used with access_policy_type of MAC whitelist.
```
Comment (kbreit) [inline plugins/modules/meraki_ms_switchport.py:357]: Please remove white list and only say allowed list instead.

**Comment by jeffkala** [comment]
@kbreit should be all set now.  Let me know if anything else is needed.

**Pull request merged**